### ISSUE TITLE
Escape the tab trap

### ISSUE THREAD

**Issue by blackfalcon** [AlaskaAirlines/auro-datepicker#23]
## Is your feature request related to a problem? Please describe.

I won't say that this is a bug, but there is not currently a way to escape this experience via the keyboard alone. 

## Describe the solution you'd like

Once a user has entered the date picker, using only the keyboard, the user cannot get out of the date picker experience. 

## Describe alternatives you've considered

In some date pickers, there is a `done` button or at the very least, there is an end to the cycle that will put the user back into the text input and close the calendar window allowing the next tab to move to another element. 

I would suggest that once the user presses the `enter` key and adds the date to the input, the calendar should close allowing the next tab to move to the next element. 

## Additional context

![tab-trap](https://user-images.githubusercontent.com/181089/179120832-39996698-52f0-41bc-ac74-f9695e0259f5.gif)

**Comment by jason-capsule42** [comment]
Current functionality is that the`esc` key will close the dropdown, move focus to the input, and then tab moves you passed the datepicker normally.

**Comment by blackfalcon** [comment]
I'd argue that this is an unexpected complexity to the end user. @leeejune I'd like some feedback on this and possibly comparative research on other similar components. We should also test this against the browser-implemented calendar UI in Chrome.

**Comment by jason-capsule42** [comment]
Note that auro-dropdown is coded to close the bib when hitting `escape` so we are following a pre-existing pattern here.

**Comment by jason-capsule42** [comment]
> > Note that auro-dropdown is coded to close the bib when hitting `escape` so we are following a pre-existing pattern here.
> 
> Sure, but that is not the only way to close it? In the example you have on the doc site, when I tab into the bib and then the button and hit the enter key, it closes. There is no secondary need to hit the `esc` key to get out of that scenario. Same for combobox. Hitting `enter` on the selected option closes the bib.

The button in that example is explicitely coded to call the `dropdown.hide()` method. The feature is coded this way so that implementers can choose if clicking on bib content, and what parts of the content, would also close the bib.

Design has stated in other conversations that keyboard users are not expected to actually use bib calendar and will instead just type a date normally. I'm not convinced we need to make any change here.
